src/orders/domain/order-detail.domain.ts

```diff
@@ -0,0 +1,31 @@
+import {
+  Entity,
+  PrimaryGeneratedColumn,
+  Column,
+  ManyToOne,
+  JoinColumn,
+} from 'typeorm';
+import { Order } from './order.domain';
+import { Product } from 'src/products/domain/product.domain';
+import { Location } from 'src/products/domain/location.domain';
+
+@Entity()
+export class OrderDetail {
+  @PrimaryGeneratedColumn('uuid')
+  id: string;
```

Comment:
Having this autogenerated ID here is not a mistake, but you could have created a composite key from the `orderId` and `productId` columns.